packages/lesswrong/server/vulcan-lib/apollo-ssr/renderPage.tsx

```diff
@@ -198,7 +198,14 @@ export const renderRequest = async ({req, user, startTime, res, clientId}: {
 
   // TODO: there should be a cleaner way to set this wrapper
   // id must always match the client side start.jsx file
-  const ssrBody = `<div id="react-app">${htmlContent}</div>`;
+  //
+  // When the theme name is "auto", we load the correct style by combining @import url()
+  // with prefers-color-scheme (see `renderJssSheetImports`). There's a long-standing
+  // Firefox bug where this can cause a flash of unstyled content. For reasons that
+  // aren't entirely obvious to me, this can be fixed by adding <script>0</script> as the
+  // first child of <body> which forces the browser to load the CSS before rendering.
+  // See https://bugzilla.mozilla.org/show_bug.cgi?id=1404468
+  const ssrBody = `<script>0</script><div id="react-app">${htmlContent}</div>`;
```

Comment:
I don't fully understand the implications of a blocking script before the content, but I'm suss of it. Can you add a condition that only inserts it if the user agent reports firefox?